pkg/agent/multicast/mcast_controller.go

```diff
@@ -181,13 +181,15 @@ func (c *Controller) clearStaleGroups() {
 	for _, obj := range c.groupCache.List() {
 		status := obj.(*GroupMemberStatus)
 		diff := now.Sub(status.lastIGMPReport)
-		if diff > c.mcastGroupTimeout {
-			// Notify worker to remove the group from groupCache if all its members are not updated before mcastGroupTimeout.
-			c.queue.Add(status.group.String())
+		if diff > c.mcastGroupTimeout && len(status.localMembers) == 0 {
```

Comment:
question: is there still meaningful to track when the last IGMP report is received and have the check here, since every member will be checked individually?

Reply:
Removed the field `lastIGMPReport` in struct `GroupMemberStatus`